solarforecastarbiter/metrics/tests/test_calculator.py

```diff
@@ -296,3 +297,47 @@ def test_apply_deterministic_bad_metric_func():
         calculator._apply_deterministic_metric_func('BAD METRIC',
                                                     pd.Series(),
                                                     pd.Series())
+
+
+@pytest.mark.parametrize('interval_label_fx, interval_label_ref', [
+    ("beginning", "beginning"),
+    ("ending", "ending"),
+    pytest.param("beginning", "ending",
+                 marks=pytest.mark.xfail(raises=ValueError)),
```

Comment:
```suggestion
                 marks=pytest.mark.xfail(raises=ValueError, strict=True)),
```

Reply:
not sure why the line isn't being hit, but it's a good idea to add `strict=True` if you know it's supposed to fail.

Reply:
Revised via https://github.com/SolarArbiter/solarforecastarbiter-core/pull/297/commits/d8abab936cacbad360ececfca09f7062b7002826

Reply:
@wholmgren Reading through the pytest docs, shouldn't we want to keep it as `strict=False` since the point of these two test cases is to check that non-matching `interval_label` values should raise a `ValueError`, but not to cause the entire test set to fail? In other words, the individual tests failing + the overall test set passing is the desired behavior.

Alternatively, I could re-write the tests using the `with pytest.raises(ValueError)` logic used in the other tests (e.g. `test_calculate_metrics()`).

Reply:
`strict=False` will cause the test to pass no matter what. `strict=True` requires that the exception be raised in order for the test to pass. 

> the entire test set to fail

You (correctly) marked up two of the individual parameter tuples, not the whole test, so I'm not sure what you mean by test set.

>with pytest.raises(ValueError)

my understanding is that this is the preferred way to assert that an exception is raised, but it's a little harder to use with `parametrize`.

Reply:
I'll try revising to use `pytest.raises(ValueError)` (probably with two separate test functions; one for tests that should pass and another for the `raises ValueError` cases).

Reply:
Just so we're on the same page... is that line hit if you step through manually?

Reply:
> Just so we're on the same page... is that line hit if you step through manually?

I may have figured out the issue. I'll push a commit and if the codecov issue goes away, then I'll explain the change. (I don't want to say anything in case I'm wrong/it's not the actual cause.)

Reply:
But to answer your question: just switching to `pytest.raises(ValueError)` didn't work (i.e. `ValueError` was not being raised by the test).

Reply:
So I don't fully understand the reasoning (yet), but it seems if I test `calculator.calculate_metrics()` with all deterministic metrics, then `ValueError` is not raised. But if I test with just the functions that require a reference forecast (which currently is just forecast skill), then `ValueError` is raised.

I'm still investigating, but at the moment, this means this PR has sufficient codecov for merging.

Reply:
Actually, the codecov may not have run on this newest commit...

Reply:
FYI you can check the coverage of the metrics tests locally by running 
```
pytest solarforecastarbiter/metrics --cov=solarforecastarbiter.metrics --cov-report=term-missing
```
You should then be able to see if the line in question was included in the tests or not.

Reply:
Thanks, @alorenzo175!